### ISSUE TITLE
authorized_key failed with "object has no attribute 'HTTPSConnection'"

### ISSUE THREAD

**Issue by mbudnick** [ansible/ansible-modules-core#2145]
##### Issue Type:
- Bug Report
##### Ansible Version:

ansible 1.9.3
  configured module search path = None
##### Ansible Configuration:

[defaults]
hostfile = /dev/null
gathering = explicit
remote_user = root
##### Environment:

running Ansible from "openSUSE 13.2 (Harlequin) (x86_64)"
managing "openSUSE 13.2 (Harlequin) (x86_64)"
##### Summary:

authorized_key failed while installing the key with

```
class CustomHTTPSConnection(httplib.HTTPSConnection):
AttributeError: 'module' object has no attribute 'HTTPSConnection'
```

If the managed machine has python installed, authorized_key can install the key without error.
##### Steps To Reproduce:

```
ansible-playbook -i dev initNewRemoteHost.yml --extra-vars "host=testserver.testnet.local" --ask-pass
```

```

---
# file: initNewRemoteHost.yml
- hosts: '{{ host }}'
  gather_facts: no
  remote_user: root

  tasks:
  - name: Add SSH key to the new remote host
    authorized_key: user=root key="{{ lookup('file', '../ssh/id_rsa.pub') }}"
```
##### Expected Results:

authorized_key install key for root successful
##### Actual Results:

```
TASK: [Add SSH key to the new remote host] ************************************
failed: [testserver.testnet.local] => {"failed": true, "parsed": false}
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1443105798.48-27107801545374/authorized_key", line 2387, in <module>
    class CustomHTTPSConnection(httplib.HTTPSConnection):
AttributeError: 'module' object has no attribute 'HTTPSConnection'
OpenSSH_6.6.1, OpenSSL 1.0.1k-fips 8 Jan 2015
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 20: Applying options for *
debug1: auto-mux: Trying existing master
debug1: mux_client_request_session: master session id: 2
Shared connection to testserver.testnet.local closed.
```
##### Workaround:

Install on the managed machine (testserver.testnet.local above) python and there will no error
or go back to ansible 1.7.1

**Comment by profhase** [comment]
Same issue -> python is installed on the machine (by default). What do you mean by 'install python'?

**Comment by mbudnick** [comment]
My managed machine is installed with less packages, so python is also not installed. After installing phython on the manged machine the task "Add SSH key to the new remote host" execute with no problem. Maybe installing the phython-package installed some dependent packages which  make the task run.

**Comment by abadger** [comment]
@mbudnick You must have python installed on the managed machine for ansible to function.  This is documented here: https://docs.ansible.com/ansible/intro_installation.html#managed-node-requirements

However, the error you're getting implies that you do have python on the managed machine.  You just that your distribution has split the ssl portion of the python stdlib into a separate package and not installed that by default.  This is not what I'd call a supported configuration but we recently added code to the 2.0 branch that should get you past that particular error (as long as you don't try to retrieve your key from an https url ;-).  https://github.com/ansible/ansible/commit/33863eb653f3ed4d6f30ab816743443f473c5eae

So I'm going to close this as fixed in 2.0.

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
